### ISSUE TITLE
Object Ownership doesn't allow Value BucketOwnerEnforced

### ISSUE THREAD

**Issue by Jlrine2** [aws-cloudformation/cfn-lint#2191]
cfn-lint version: 0.56.3

When linting a CF temptate containing a bucket with bucket ownership set to `ObjectOwnership: BucketOwnerEnforced` linking fails with the error `
E3030 You must specify a valid value for ObjectOwnership (BucketOwnerEnforced). Valid values are ["ObjectWriter", "BucketOwnerPreferred"]` however Cloudformation allows the value `BucketOwnerEnforced` per the [documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-ownershipcontrolsrule.html)

sample template 

```yml
  LogBucket:
    Type: AWS::S3::Bucket
    Properties:
      PublicAccessBlockConfiguration:
        BlockPublicAcls: True
        IgnorePublicAcls: True
        BlockPublicPolicy: True
        RestrictPublicBuckets: True
      BucketEncryption:
        ServerSideEncryptionConfiguration:
          - BucketKeyEnabled: true
      OwnershipControls:
        Rules:
          - ObjectOwnership: BucketOwnerEnforced
```

Cfn-lint uses the [CloudFormation Resource Specifications](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-resource-specification.html) as the base to do validation. These files are included as part of the application version. Please update to the latest version of `cfn-lint` or update the spec files manually (`cfn-lint -u`)

**Comment by PatMyron** [comment]
will get picked up in https://github.com/aws-cloudformation/cfn-lint/pull/2188 once the [resource provider schema is updated for `us-east-1`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/resource-type-schemas.html)

**Comment by kddejong** [comment]
I'm waiting on an issue in the spec to be resolved before I can get this merged. Should be resolved this week.

**Comment by kddejong** [comment]
This is coming from the definition in the public registry.  I'm looking into why there is a mix match between documentation and the resource schema definition.
```json
"OwnershipControlsRule": {
            "type": "object",
            "additionalProperties": false,
            "properties": {
                "ObjectOwnership": {
                    "description": "Specifies an object ownership rule.",
                    "type": "string",
                    "enum": [
                        "ObjectWriter",
                        "BucketOwnerPreferred"
                    ]
                }
            }
        },
```

**Comment by PatMyron** [comment]
Upgrading should pick this up now

**Comment by asjohnston-asf** [comment]
FYI, earlier this week I confirmed the project @Jlrine2 and I were working on no longer encounters this error as of cfn-lint v0.58.2.
